### DISCUSSION TITLE
Error with offsetWidth Reading During Screen Width Change and Layout Transition

### DISCUSSION THREAD

**Discussion by jszufa** [vuepress/ecosystem#95] [Q&A] [answered]
Description:
I am encountering an issue where an Navbar&offsetWidth related error appears when changing the screen width, but this only occurs after changing the layout from one to another (e.g., from Home to Details). The specific error message is: "Cannot read properties of null (reading 'offsetWidth')". This issue arises in a VuePress 2 project and seems to be linked to a transition between different layouts. The error does not affect the page's functionality directly but is consistently reproducible under the specified conditions. It appears in the console as a TypeError, suggesting that the script is attempting to access offsetWidth of an element that is not currently mounted or has been removed from the DOM.

Error: Cannot read properties of null (reading 'offsetWidth')
TypeError: Cannot read properties of null (reading 'offsetWidth')
    at eval (webpack-internal:///./node_modules/esbuild-loader/dist/index.cjs??clonedRuleSet-281.use[0]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use[0]!./node_modules/@vuepress/theme-default/lib/client/components/Navbar.vue?vue&type=script&setup=true&lang=ts:43:53)
    at eval (webpack-internal:///./node_modules/@vuepress/theme-default/lib/client/composables/useUpdateDeviceStatus.js:29:49)

Steps to Reproduce:

Create a global layout that includes a default navbar component. This global layout is applied across different views or layouts in the application (e.g., Home and Details).
Start with the application in a specific layout (e.g., Home).
Switch to another layout (e.g., Details) via the navigation or routing.
Resize the browser window or change the screen width.
Observe the error in the console related to offsetWidth.
Expected Behavior:
Changing layouts and subsequently resizing the window should not produce any errors related to offsetWidth.

Actual Behavior:
An error is thrown in the console when the screen width is adjusted after changing layouts, indicating a problem with accessing offsetWidth of a null element.

**Comment by Mister-Hope** [comment, marked as answer]
I believe this is fixed in https://github.com/vuepress/ecosystem/commit/03e67c11e8388041c67fb0d5933e632db655e701, will be roll out in next version
